packages/front-end/hooks/useDownloadDataExport.ts

```diff
@@ -46,10 +47,21 @@ export const useDownloadDataExport = ({
     setIsDownloading(true);
 
     apiCall(url)
-      .then((response: DataExportFileResponse) => {
-        saveAs({ textContent: response.data, fileName: response.fileName });
+      .then((response: DataExportFileResponse | ApiErrorResponse) => {
```

Comment:
I'm pretty sure `apiCall` will never return an `ApiErrorResponse` (if it is, something is wrong).  It should throw an exception instead using `response.message`.  So I think the `.catch` you have below already handles this case.

Reply:
Unfortunately, it falls into the .`then()` block when the server returns a response with a failed status code, which is why I needed to add this. I, too, was expecting it to fall into the `.catch()` block but it does not. [Here](https://github.com/growthbook/growthbook/blob/main/packages/front-end/services/auth.tsx#L251-L253) is the code. 

But it looks like if I add `status: 403` to the payload it'll fall into the catch block. But then I couldn't use `ApiErrorResponse` I'd have to use something else or just inline the type. I can try that even though it should've failed. But that's the backend compensating for a bug on the front-end if I do.

Reply:
In 67c17a0332783ccdbbb0fffc9d569f0adbc7d7a8 I've added a new type `PrivateApiErrorResponse` which includes the status number so that the `_makeApiCall` method fails properly.